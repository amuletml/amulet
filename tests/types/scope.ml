external val (+) : int -> int -> int =
  "function(x, y) return x + y end"

module X = begin
  type t = T
  let x = T
end

let _ =
  let _ = X.x + 1
  (* local opens unqualify *)
  let _ =
    let open X
    x + 1
  let _ = X.( x + 1 )

  (* make sure that the unqualified scope doesn't leak *)
  let _ = X.x + 1
  ()
