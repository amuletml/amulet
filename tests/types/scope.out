scope.ml[10:11 ..10:13]: error (E2001)
   │ 
10 │   let _ = X.x + 1
   │           ^^^
  Couldn't match actual type X.t
    with the type expected by the context, int
scope.ml[14:5 ..14:5]: error (E2001)
   │ 
14 │     x + 1
   │     ^
  Couldn't match actual type t
    with the type expected by the context, int
scope.ml[15:15 ..15:15]: error (E2001)
   │ 
15 │   let _ = X.( x + 1 )
   │               ^
  Couldn't match actual type t
    with the type expected by the context, int
scope.ml[18:11 ..18:13]: error (E2001)
   │ 
18 │   let _ = X.x + 1
   │           ^^^
  Couldn't match actual type X.t
    with the type expected by the context, int
